### PR TITLE
Be more compliant!

### LINKED ISSUE #12: Be compliant with data store convention
Achive full compliancy with https://github.com/dcs4cop/xcube/issues/330.

### PR DESCRIPTION
**Pull request by forman**
* Documented store open parameters. 
* Store open parameter `variable_names` and  `CubeConfig` parameter `band_names` may now be `None`. In this case all variables/bands are included. 

Closes #12

### CODE REVIEW

**Review by dzelge — APPROVED** [review]

File: xcube_sh/chunkstore.py [hunk 1]
```diff
@@ -151,6 +151,8 @@ def __init__(self,
         self._add_static_array('time', t_array, time_attrs)
         self._add_static_array('time_bnds', t_bnds_array, time_bnds_attrs)
 
+
```
Comment (TonioF) [inline xcube_sh/chunkstore.py:154]: Why so many blank lines? I think one was enough.

File: test/test_chunkstore.py [hunk 2]
```diff
@@ -191,6 +189,22 @@ def test_tiles_to_time_ranges(self):
                          [(tr[0].isoformat(), tr[1].isoformat()) for tr in time_ranges])
 
 
+class SentinelHubStore3DTestWithAllBands(SentinelHubStoreTest):
+    def get_cube_config(self):
+        return CubeConfig(dataset_name='S2L2A',
+                          geometry=(10.2, 53.5, 10.3, 53.6),
```
Comment (TonioF) [inline test/test_chunkstore.py:195]: Would you explicitly set band_names to None here?

File: test/test_chunkstore.py [hunk 3]
```diff
@@ -28,10 +27,9 @@
 import xarray as xr
 import zarr
 
-from xcube_sh.config import CubeConfig
-from xcube_sh.metadata import SentinelHubMetadata
-from xcube_sh.sentinelhub import SerializableOAuth2Session
 from xcube_sh.chunkstore import SentinelHubChunkStore
+from xcube_sh.config import CubeConfig
+from xcube_sh.metadata import SentinelHubMetadata, S2_BAND_NAMES
```
Comment (TonioF) [inline test/test_chunkstore.py:32]: Please split this into two lines

File: CHANGES.md [hunk 4]
```diff
@@ -1,8 +1,11 @@
 ## Changes in 0.3 (in dev)
 
+* Documented store open parameters. Store open parameter `variable_names` and  
+  `CubeConfig` parameter `band_names` may now be `None`. In this case all 
+  variables/bands are included. (#12)
 * `CubeConfig` now accepts parameter `bbox` instead of parameter `geometry`.
-   `geometry` is deprecated, but will be supported later. It may then 
-   also be a WKG Geometry string.
```
Comment (TonioF) [inline CHANGES.md:5]: WKT

**Review by TonioF — CHANGES_REQUESTED** [review]
All recommended changes are minor. Changes look good!

File: test/test_chunkstore.py (hunk 2 above)
Comment (pont-us) [inline test/test_chunkstore.py:195]: Shouldn't `geometry` be `bbox`? (Unless of course you're deliberately testing the deprecated keyword.)

File: xcube_sh/chunkstore.py (hunk 1 above)
Comment (pont-us) [reply xcube_sh/chunkstore.py:154]: PEP 8 agrees with you ("Use blank lines in functions, sparingly...").

File: xcube_sh/store.py [hunk 5]
```diff
@@ -74,6 +79,36 @@ def get_open_data_params_schema(self, data_id: str = None) -> JsonObjectSchema:
...
+            If not given, all variables are included.
+        * ``variable_units: Union[str, Sequence[str]]`` - units for all or each variable
+        * ``variable_sample_types: Union[str, Sequence[str]]`` - sample types for all or each variable
+        * ``crs: str`` - spatial CRS identifier. must be a valid OGC CRS URI.
+        * ``tile_size: Tuple[int, int]`` - optional tuple of spatial tile sizes in pixels.
+        * ``bbox: Tuple[float, float, float, float]`` - spatial coverage given as (minx, miny, maxx, maxy)
+            in units of the CRS. Required parameter.
+        * ``spatial_res: float`` - spatial resolution in unsits of the CRS^.
+            Required parameter.
+        * ``time_range: Tuple[Optional[str], Optional[str]]`` - tuple (start-time, end-time).
```
Comment (pont-us) [inline xcube_sh/store.py:98]: ISO 8601 can probably be assumed, but might be as well to make it explicit.

**Review by pont-us — CHANGES_REQUESTED** [review]
Some minimal changes suggested.

File: test/test_chunkstore.py (hunk 2 above)
Comment (forman) [reply test/test_chunkstore.py:195]: That is redundant because `band_names` is optional _and_ documented.

Comment (forman) [reply test/test_chunkstore.py:195]: Right!

File: xcube_sh/store.py (hunk 5 above)
Comment (forman) [reply xcube_sh/store.py:101]: What do you want me to do?

File: xcube_sh/chunkstore.py (hunk 1 above)
Comment (forman) [reply xcube_sh/chunkstore.py:154]: It should be 1 line only.

File: xcube_sh/store.py (hunk 5 above)
Comment (pont-us) [reply xcube_sh/store.py:101]: Explicitly state the format of the start-time and end-time strings and the semantics of an omitted value for either or both of them. (Or link/refer to the store conventions document for details on this.)

File: xcube_sh/store.py [hunk 6]
```diff
@@ -149,17 +191,21 @@ def _get_open_data_params_schema(self, dsd: DataDescriptor = None) -> JsonObject
...
                                               JsonStringSchema(format='date-time'))),
             # TODO: add pattern
-            time_period=JsonStringSchema(default='1D'),
-            time_tolerance=JsonStringSchema(default=DEFAULT_TIME_TOLERANCE),
+            time_period=JsonStringSchema(default='1D', nullable=True,
+                                         enum=[None,
+                                               *map(lambda n: f'{n}D', range(1, 7)),
+                                               '1W', '2W']),
+            time_tolerance=JsonStringSchema(
+                default=DEFAULT_TIME_TOLERANCE),
```
Comment (TonioF) [inline xcube_sh/store.py:199]: As it is documented that you want a pandas-compatible format, you can leave this as it is, but I'd much appreciate it if you could add an enum (as in the time_period) or a pattern.

**Review by TonioF — APPROVED** [review]
Mostly approved, just one comment.

**Review by pont-us — APPROVED** [review]
Added one suggestion about time format descriptions in docstring, approving nevertheless.

File: xcube_sh/store.py (hunk 5 above)
Comment (forman) [reply xcube_sh/store.py:101]: I fully agree!

File: xcube_sh/store.py (hunk 6 above)
Comment (forman) [reply xcube_sh/store.py:199]: I'll do!

**Pull request merged**